sdk/include/opentelemetry/sdk/logs/logger.h

```diff
@@ -57,13 +61,18 @@ class Logger final : public opentelemetry::logs::Logger
            opentelemetry::trace::TraceFlags trace_flags,
            opentelemetry::common::SystemTimestamp timestamp) noexcept override;
 
+  /** Returns the associated instruementation library */
+  const opentelemetry::sdk::instrumentationlibrary::InstrumentationLibrary &
+  GetInstrumentationLibrary() const noexcept;
+
 private:
   // The name of this logger
   std::string logger_name_;
 
   // The logger provider of this Logger. Uses a weak_ptr to avoid cyclic dependency issues the with
   // logger provider
   std::weak_ptr<LoggerProvider> logger_provider_;
+  std::shared_ptr<instrumentationlibrary::InstrumentationLibrary> instrumentation_library_;
```

Comment:
I understand this is coming from tracer implementation, but now thinking why do store it as `shared_ptr` instead of unique_ptr, or as value. We are anyway passing it to exporters as reference.